app/assets/javascripts/snowcrash/modules/issues/table.js.coffee

```diff
@@ -1,218 +1,31 @@
-          $that.prop('checked', false)
-          issue_id = $that.val()
-
-          $($row.find('td')[that.tagColumnIndex]).replaceWith(data.tag_cell)
-          $("#issues #issue_#{issue_id}").replaceWith(data.issue_link)
-          if $(@selectedIssuesSelector).length == 0
-            that.resetToolbar()
-
-        error: (foo,bar,foobar) ->
-          $($row.find('td')[that.tagColumnIndex]).replaceWith("<td class='text-error'>Please try again</td>")
-      }
-
-  onMergeSelected: (event) =>
-    url = $(event.target).data('url')
-    issues_to_merge = []
-
-    $('.js-tbl-issues').find(@selectedIssuesSelector).each ->
-      $row = $(this).parent().parent()
-
-      id = @.name.split('_')[1]
-      issues_to_merge.push(id)
-
-    location.href = "#{url}?ids=#{issues_to_merge}"
-
-  saveColumnState: ->
-    if Storage?
-      localStorage.setItem(@storageKey(), JSON.stringify(@selectedColumns))
-    else
-      console.log "The browser doesn't support local storage of settings."
-      console.log "Column selection can't be saved."
-
-  showHideColumns: =>
-    that = this
-
-    $.each @$table.find('thead th'), (index, th)->
-      $th = $(th)
-
-      if (column = $(th).data('column'))
-        that.tagColumnIndex ||= index if column == 'tags'
-        if that.selectedColumns.indexOf(column) > -1
-          that.$table.find("td:nth-child(#{index + 1})").css('display', 'table-cell')
-          $th.css('display', 'table-cell')
-        else
-          that.$table.find("td:nth-child(#{index + 1})").css('display', 'none')
-          $th.css('display', 'none')
-
-  storageKey: ->
-    @project ||= $('.brand').data('project')
-    "project.#{@project}.issue_columns"
-
 jQuery ->
   if $('body.issues.index').length
+    indexTable = new IndexTable('issue')
 
-    # Checkbox behavior: select all, show 'btn-group', etc.
-    $('.js-select-all-issues').click (e)->
-      $allCheckbox = $(this).find('input[type=checkbox]')
-      isChecked = $allCheckbox.prop('checked')
-      if e.target != $allCheckbox[0]
-        isChecked = !isChecked
-        $allCheckbox.prop('checked', isChecked)
-
-      $('input[type=checkbox].js-multicheck:visible').prop('checked', isChecked)
+    onMergeSelected = (event) =>
+      url = $(event.target).data('url')
+      issues_to_merge = []
 
-    # when selecting standalone issues, check if we must also check 'select all'
-    $('input[type=checkbox].js-multicheck').click ->
-      _select_all = $(this).prop('checked')
+      $('.js-items-table').find(indexTable.selectedItemsSelector).each ->
+        $row = $(this).parent().parent()
 
-      if _select_all
-        $('input[type=checkbox].js-multicheck').each ->
-          _select_all = $(this).prop('checked')
-          _select_all
+        id = @.name.split('_')[2]
+        issues_to_merge.push(id)
 
-      $('.js-select-all-issues > input[type=checkbox]').prop('checked', _select_all)
+      location.href = "#{url}?ids=#{issues_to_merge}"
 
     refreshToolbar = ->
```

Comment:
`refreshToolbar` is also defined in `IndexTable` class, and here without a class
I think that creating a class `IssuesTable` that inherits from `IndexTable` makes sense then.

Reply:
✅